packages/alfa-dom/src/node/element.ts

```diff
@@ -262,6 +262,47 @@ export class Element extends Node implements Slot, Slotable {
     return path;
   }
 
+  /**
+   * Return all descendants of this element's root whose id is listed in an IDlist attribute (headers, aria-labelledby, …)
+   */
+  public resolveAttributeReferences(
+    name: string,
+    options: Node.Traversal = {}
+  ): Array<Element> {
+    return this.root(options).resolveReferences(
+      ...this.attribute(name)
+        .map((attribute) => attribute.tokens())
+        .getOr([])
+    );
+  }
+
+  /**
+   * @see https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#enumerated-attribute
+   */
+  public enumerateAttribute(
+    name: string,
+    ...keywords: Array<string>
+  ): string | Element.EnumeratedAttributeError {
+    const attribute = this.attribute(name);
+
+    if (attribute.isNone() || attribute.get().value === "")
+      return Element.EnumeratedAttributeError.Missing;
+
+    const value = attribute.get().value.toLowerCase();
+
+    return keywords.includes(value)
+      ? value
+      : Element.EnumeratedAttributeError.Invalid;
+  }
```

Comment:
As mentioned in https://github.com/Siteimprove/alfa/pull/197#discussion_r408673970, this belongs in `Attribute` as also made evident by the fact that the method is now postfixed with, well, `Attribute` 🙈 I do like this way of passing a predefined set of valid keywords though, so I'm thinking an overload would make sense:

```ts
Attribute#enumerate(): Option<string>;
Attribute#enumerate(valid: string, ...rest: Array<string>): Option<string>;
```

Reply:
The problem I had when trying to put it on `Attribute` is that it cannot handle the case where the attribute is missing, so the full `if (attribute.isNone())` part must still be done at the `Element` level. Either requiring a specific helper (like this one), or the need to systematically chain:
`element.attribute("foo").map(attribute => attribute.enumerate()).getOr(Err.of("missing"))`
instead of 
`element.enumerateAttribute("foo")`

Might still be better to have the helper split in two, however…

Reply:
The chaining approach is indeed what should be used; that's the whole reason for the monadic approach:

```ts
const foo: string = element
  .attribute("foo")
  .map(attribute => 
    attribute.enumerate("bar", "baz").getOr("invalid")
  )
  .getOr("missing");
```

If I don't care about missing and invalid states, it's as simple as:

```ts
const foo: Option<string> = element
  .attribute("foo")
  .flatMap(attribute => attribute.enumerate("bar", "baz"));
```

If I only care about missing states, I can instead do:

```ts
const foo: Option<string> = element
  .attribute("foo")
  .map(attribute => attribute.enumerate("bar", "baz"))
  .getOr(Some.of("missing"));
```

By not making too many assumptions about what the caller wants to do, we've now made it possible to model a lot more things than if we had only designed the API with a single use case in mind.